crates/sui-types/src/passkey_authenticator.rs

```diff
@@ -0,0 +1,191 @@
+                error: "get client data".to_string(),
+            })?;
+        let parsed_challenge =
+            Base64UrlUnpadded::decode_vec(&client_data_json.challenge).map_err(|e| {
+                SuiError::InvalidSignature {
+                    error: e.to_string(),
+                }
+            })?;
+        let mut hasher = DefaultHash::default();
+        hasher.update(&bcs::to_bytes(&intent_msg).expect("Message serialization should not fail"));
+        let digest = hasher.finalize().digest;
+
+        // check client_data.challenge == sha256(intent_msg(tx))
+        if parsed_challenge != digest {
+            return Err(SuiError::InvalidSignature {
+                error: "invalid challenge".to_string(),
+            });
+        };
+
+        // msg = authenticator_data || sha256(client_data)
+        let mut message = self.authenticator_data.clone();
+        let client_data_hash = Sha256::digest(self.client_data.as_slice()).digest;
+        message.extend_from_slice(&client_data_hash);
+
+        match self.user_signature.scheme() {
+            SignatureScheme::Secp256r1 => {
+                let pk = Secp256r1PublicKey::from_bytes(self.user_signature.public_key_bytes())
+                    .map_err(|_| SuiError::InvalidSignature {
+                        error: "Invalid r1 pk bytes".to_string(),
+                    })?;
+                if author
+                    != SuiAddress::from(
+                        &PublicKey::try_from_bytes(
+                            SignatureScheme::Secp256r1,
+                            self.user_signature.public_key_bytes(),
+                        )
+                        .unwrap(),
+                    )
+                {
+                    return Err(SuiError::InvalidSignature {
+                        error: "Invalid author".to_string(),
+                    });
+                };
+                let sig = Secp256r1Signature::from_bytes(self.user_signature.signature_bytes())
+                    .map_err(|_| SuiError::InvalidSignature {
+                        error: "Invalid r1 signature bytes".to_string(),
+                    })?;
+
+                pk.verify(&message, &sig)
+                    .map_err(|_| SuiError::InvalidSignature {
+                        error: "verify failed".to_string(),
+                    })
+            }
+            _ => Err(SuiError::InvalidSignature {
+                error: "Invalid signature scheme for passkey".to_string(),
+            }),
+        }
+    }
+}
+
+impl ToFromBytes for PasskeyAuthenticator {
+    fn from_bytes(bytes: &[u8]) -> Result<Self, FastCryptoError> {
+        // The first byte matches the flag of MultiSig.
+        if bytes.first().ok_or(FastCryptoError::InvalidInput)?
+            != &SignatureScheme::PasskeyAuthenticator.flag()
+        {
+            return Err(FastCryptoError::InvalidInput);
+        }
+        let zk_login: PasskeyAuthenticator =
```

Comment:
why is this called zk_login?

Reply:
i think you are reading an older commit, latest is on: 1b912aa0af54f125fcf04c0e765abb1a849b6a08